gum_test.html

```diff
@@ -287,6 +312,38 @@ <h4>getUserMedia is missing!</h4>
       setTimeout(captureImage, 2000);
     }
   }
+
+  function startEqualizer(stream) {
+    const audioCtx = new AudioContext();
+    const analyser = audioCtx.createAnalyser();
+    audioCtx.createMediaStreamSource(stream).connect(analyser);
+
+    const canvasCtx = equalizer.getContext("2d");
+    const data = new Uint8Array(equalizer.width);
+
+    let interval = setInterval(() => {
+      if (stream.getAudioTracks()[0].readyState != "live") {
+        clearInterval(interval);
+        return;
+      }
+      canvasCtx.fillStyle = "#ffffff";
+      canvasCtx.clearRect(0, 0, equalizer.width, equalizer.height);
+      analyser.getByteFrequencyData(data);
+      canvasCtx.lineWidth = 2;
+      let x = 0;
+      for (let y of data) {
+        y = equalizer.height - (y / 128) * equalizer.height / 4;
+        let c = Math.floor((x*255)/equalizer.width);
+        canvasCtx.fillStyle = "rgb("+c+",0,"+(255-x)+")";
+        canvasCtx.fillRect(x, y, 2, equalizer.height - y)
+        x++;
```

Comment:
Have you made sure x doesn't overshoot the canvas width? That would be unnecessary. You can control how many frequency buckets you get from the analyser with the `fftSize` attribute.